dart/lib/src/transport/http_transport.dart

```diff
@@ -68,26 +77,49 @@ class HttpTransport implements Transport {
     } else {
       _options.logger(
         SentryLevel.debug,
-        'Event ${event.eventId} was sent successfully.',
+        'Envelope ${envelope.header.eventId ?? "--"} was sent successfully.',
       );
     }
 
     final eventId = json.decode(response.body)['id'];
     return eventId != null ? SentryId.fromId(eventId) : SentryId.empty();
   }
 
-  List<int> _bodyEncoder(
-    Map<String, dynamic> data,
-    Map<String, String> headers, {
-    required bool compressPayload,
-  }) {
-    // [SentryIOClient] implement gzip compression
-    // gzip compression is not available on browser
-    var body = utf8.encode(json.encode(data));
-    if (compressPayload) {
-      body = compressBody(body, headers);
+  Future<StreamedRequest> _createStreamedRequest(
+      SentryEnvelope envelope) async {
+    final streamedRequest = StreamedRequest('POST', _dsn.postUri);
+
+    if (_options.compressPayload) {
+      final envelopeData = <int>[];
+      await envelope.envelopeStream().forEach(envelopeData.addAll);
```

Comment:
We're buffering the whole data in memory here.
Is it possible to implement the compression bits on top of Stream instead?

This way we can decorate the streams and just pipe data through to the network without the need to allocate everything in memory.

i.e:

https://github.com/getsentry/sentry-dotnet/blob/a9304a0a4b4702d0e62e2703d55c66483d27a0e5/src/Sentry/Internal/Http/GzipRequestBodyHandler.cs#L85-L94

Reply:
Ok, it seems that the encoder offered from `GzipCodec` can handle streaming, just wrote a small unit test which works. Will change this. https://api.dart.dev/stable/2.10.4/dart-io/ZLibEncoder/startChunkedConversion.html

Reply:
Ok, this seems to work. \o/  Thanks for bringing it up, i wrongly assumed we have to gzip the complete payload.

Reply:
we also need to be sure this works on Web though